src/main/kotlin/com/openlattice/indexing/IndexingService.kt

```diff
@@ -90,9 +89,9 @@ class IndexingService(
 
     @Suppress("UNUSED")
     private val indexingWorker = executor.submit {
-        Stream.generate { indexingQueue.take() }
-                .parallel().forEach { entitySetId ->
-                    try {
+        while ( true ) {
```

Comment:
This moves the queue polling into series but it also moves the work itself into series, which probably isn't helpful. Can the rest of this loop be submitted into work queue, either via FJP or something else?

Reply:
We can definitely do that. For context, this queue is only ever populated by manually calling `IndexingController.reindex` so I figured serial, executor-thread blocking would be fine here.

Reply:
Yeah that's probably ok.